### ISSUE TITLE
Remove ArmijoShooting.py file

### ISSUE THREAD

**Issue by msparapa** [Rapid-Design-of-Systems-Laboratory/beluga#120]
I believe in update #110 I was able to implement the method used in `ArmijoShooting.py` to the standard `Shooting.py` with setting `algorithm=Armijo`. Seemed to converge decently. Can you confirm @SeanMatthewNolan that I implemented this correctly? If so we can get rid of the extra files.

**Comment by SeanMatthewNolan** [comment]
@msparapa I just checked it, and it all looks good to me.

**Comment by SeanMatthewNolan** [comment]
Actually, the check for a small lambda (in this case ll), should be in the while loop. In essence, I think it should look like this:

`while (r_try >= (1-a*ll) * err) and (r_try > self.tolerance):`
`    step = ll*dy0`
`    res_try = _constraint_function_wrapper(Xinit + step)`
`    r_try = np.linalg.norm(res_try)`
`    ll *= reduct`
`    if ll <= 0.1:`
`        break`

This is to prevent really the step size getting extremely small and to just move forward. I found this helpful because of the numerical issues.

**Comment by msparapa** [comment]
Huh I definitely thought I put that in the while loop statement itself as

```
while (r_try >= (1-a*ll) * err) and (r_try > self.tolerance) and (ll > 0.1):
```

but I guess I had merged the changes too soon. Is there any reason to use a break within the loop as opposed to in the while statement like I have above?

**Comment by SeanMatthewNolan** [comment]
No, I agree, putting it in the while statement is probably better. It was that way just because I added it later.

**Comment by msparapa** [comment]
Ok cool. I can do this later tonight.

**Comment by msparapa** [comment]
Do you think we should also remove the `algorithm=traditional` method from the shooting solver?

1. It's not even the original traditional method, we have to use beta=1 to get anything to converge.
2. I've noticed Armijo has significantly better convergence for pretty much every problem I've tried.

When implementing #120, I think the current `traditional` should be removed, `Armijo` is made to be the new default, and also I think a check should be added so if the `algorithm` is a function and not a string, it'll assume its some optimization algorithm of the same format as `scipy.optimize.minimize`. That will leave the capabilities open ended.

**Comment by SeanMatthewNolan** [comment]
I agree. The fact the the Armijo method is almost always better is why I put it in in the first place. I can't see myself wanting the old one.

**Comment by msparapa** [comment]
Fixed this, and also added some of SciPy's root solvers (`optimize.root` and `optimize.fsolve`). @SeanMatthewNolan I deleted your planar hypersonic `.py` files since I didn't think they were needed anymore. I had a bit of difficulty with planar hypersonic so I have that using `SLSQP` for the time being, but I'll probably rewrite some of the continuation steps at a later date since they we're originally made on the old method.
